packages/@unimodules/core/ios/UMCore/UMAppDelegateWrapper.m

```diff
@@ -124,6 +124,17 @@ - (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserAct
   return result;
 }
 
+#pragma mark - BackgroundSession
+- (void)application:(UIApplication *)application handleEventsForBackgroundURLSession:(NSString *)identifier completionHandler:(void (^)(void))completionHandler
+{
+  SEL selector = @selector(application:handleEventsForBackgroundURLSession:completionHandler:);
+  NSArray<id<UIApplicationDelegate>> *subcontractorsArray = [self getSubcontractorsImplementingSelector:selector];
+  
+  for (id<UIApplicationDelegate> subcontractor in subcontractorsArray) {
+    [subcontractor application:application handleEventsForBackgroundURLSession:identifier completionHandler:completionHandler];
```

Comment:
I'm just wondering, have you check what happens when we call the completion handler more than once? I guess we won't have more subcontractors but it might be worth to note and know about this.

Reply:
I've tried to invoke compilation handler twice... and nothing happened.

Reply:
However, in the future, we should ensure that the compilation handler is called only when all subcontractors are done.

Reply:
I second @tsapeta's comment—even though this may not be a serious problem, in theory we would like OS to act as if we handled all events only… once we handle all events. We already had to solve this in `EXNotificationCenterDelegate`, maybe it makes sense to just copy it here too?

 https://github.com/expo/expo/blob/fb20f53db63d27f9c2657ef8d2f2abfd23c88b8b/packages/expo-notifications/ios/EXNotifications/Notifications/EXNotificationCenterDelegate.m#L141-L167